ansible_base/api_documentation/postprocessing_hooks.py

```diff
@@ -0,0 +1,435 @@
+        # Simple resources need singularization
+        return f"{operation_name} {singularize(resource_name)}"
+
+    return None  # Indicate that this isn't a standard CRUD operation
+
+
+def generate_custom_action_description(operation_name, resource_name, operation):
+    """Generate description for custom actions."""
+    base_description = operation.get('description', '').strip()
+    clean_desc = clean_base_description(base_description)
+    return f"{operation_name} {clean_desc}" if clean_desc else f"{operation_name} {resource_name}"
+
+
+def clean_base_description(description):
+    """Clean up base description by removing common boilerplate."""
+    if not description:
+        return ""
+
+    clean_desc = description.strip()
+
+    # Remove common prefixes
+    prefixes_to_remove = [
+        'API endpoint that allows',
+        'API endpoint that',
+        'API endpoint for',
+        'A view class for managing and displaying',
+        'Endpoint:',
+    ]
+
+    for prefix in prefixes_to_remove:
+        if clean_desc.lower().startswith(prefix.lower()):
+            clean_desc = clean_desc[len(prefix) :].strip()
+            break
+
+    # Remove trailing period
+    if clean_desc.endswith('.'):
+        clean_desc = clean_desc[:-1]
+
+    # If description is too long or contains multiple sentences, truncate
+    # This handles cases like settings_getter where entire docstring was used
+    if len(clean_desc) > 100 or '\n' in clean_desc:
+        # Take first sentence or first 100 chars
+        first_line = clean_desc.split('\n')[0]
+        if '.' in first_line[:100]:
+            clean_desc = first_line.split('.')[0]
+        else:
+            clean_desc = first_line[:100]
+
+    return clean_desc.strip()
+
+
+def _should_skip_ai_description(operation_id):
+    """Check if an operation should skip AI description generation."""
+    if not operation_id:
+        return False
+
+    prefix = operation_id.split('_')[0] if '_' in operation_id else operation_id
+    if prefix in SKIP_AI_DESCRIPTION_PREFIXES:
+        return True
+    return False
+
+
+def _lookup_resource_purpose(operation_id):
+    """Look up resource_purpose for an operation via its prefix and ViewSet class."""
+    prefix = extract_operation_prefix(operation_id)
+    class_info = OPERATION_CLASS_MAP.get(prefix)
+
+    if not class_info:
+        return None, None
+
+    class_name, _, _ = class_info  # Extract class name from (class_name, path_parts_count, path_parts) tuple
+    resource_purpose = RESOURCE_PURPOSE_MAP.get(class_name)
+    return resource_purpose, class_name
+
+
+def _generate_from_resource_purpose(resource_purpose, action):
+    """Generate description from resource_purpose if action is supported."""
+    if not resource_purpose:
+        return None
+
+    if action in OPERATION_DESCRIPTION_TEMPLATES.keys():
+        return generate_description_from_purpose(resource_purpose, action)
+
+    return None
```

Comment:
`generate_description_from_purpose` returns resource_purpose if it doesn't get a template based on the action so it seems like this could just be something like:
```
return generate_description_from_purpose(resource_purpose, action) if action in OPERATION_DESCRIPTION_TEMPLATES.keys() else None
```
Unless the reource_purpose could be None or '' but even then you could consense this to be 
```
    if resource_purpose and action in OPERATION_DESCRIPTION_TEMPLATES.keys():
        return generate_description_from_purpose(resource_purpose, action)

    return None
```